src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -3695,15 +3690,15 @@ public void handleLabelsFileUpload(FileUploadEvent event) {
             try {
                 uploadStream = file.getInputstream();
             } catch (IOException ioex) {
-                logger.warning("the file " + file.getFileName() + " failed to upload!");
+                logger.log(Level.WARNING, "the file {0} failed to upload!", file.getFileName());
```

Comment:
Is there a way to use the interpolation without logger.log(Level.WARNING, which is just a bit clunky?

Reply:
No. This was discussed before - there a nontrivial runtime cost for composing a string, and if you send a message to the logger and then it's ignored it's just a waste. The form used here is the current "best practice", and was actually created by a NetBeans' code hint.